paddle/go/cmd/master/master.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"net/rpc"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/PaddlePaddle/Paddle/paddle/go/master"
+	"github.com/wangkuiyi/recordio"
+)
+
+const (
+	taskTimeoutDur = 20 * time.Minute
+	taskTimeoutMax = 3
+)
+
+func main() {
+	port := flag.Int("p", 0, "port of the master server")
+	dataset := flag.String("d", "", "dataset: comma separated path to RecordIO files")
+	faultTolerant := flag.Bool("fault-tolerance", false, "enable fault tolerance (requires etcd).")
+	flag.Parse()
```

Comment:
@typhoonzero @wangkuiyi Done. Using https://github.com/namsral/flag, it's a drop-in replacement for `flag`, adding support for environment variable and plain config file. I think https://github.com/spf13/viper probably is an overkill for the current situation (viper supports loading and monitoring config etcd, ...) And it have different syntax than the offical `flag` package, harder to learn / maintain.